contracts/SpaceSplitOperation.sol

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright ©️ 2018 Galt•Space Society Construction and Terraforming Company
+ * (Founded by [Nikolai Popeka](https://github.com/npopeka),
+ * [Dima Starodubcev](https://github.com/xhipster),
+ * [Valery Litvin](https://github.com/litvintech) by
+ * [Basic Agreement](http://cyb.ai/QmSAWEG5u5aSsUyMNYuX2A2Eaz4kEuoYWUkVBRdmu9qmct:ipfs)).
+ *
+ * Copyright ©️ 2018 Galt•Core Blockchain Company
+ * (Founded by [Nikolai Popeka](https://github.com/npopeka) and
+ * Galt•Space Society Construction and Terraforming Company by
+ * [Basic Agreement](http://cyb.ai/QmaCiXUmSrP16Gz8Jdzq6AJESY1EAANmmwha15uR3c1bsS:ipfs)).
+ */
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+//pragma experimental ABIEncoderV2;
+
+import "./utils/WeilerAtherton.sol";
+import "./utils/PolygonUtils.sol";
+import "./SpaceToken.sol";
+import "./SplitMerge.sol";
+
+contract SpaceSplitOperation {
+  using WeilerAtherton for WeilerAtherton.State;
+
+  WeilerAtherton.State private weilerAtherton;
+
+  // TODO: use stages
+  enum Stage {
+    NONE,
+    CONTRACT_INIT,
+    POLYGONS_PREPARE,
+    POLYGONS_INIT,
+    SEGMENTS_ADD,
+    MARTINEZ_RUEDA_PROCESS,
+    INTERSECT_POINTS_ADD,
+    WEILER_ATHERTON_BUILD,
+    POLYGONS_FINISH
+  }
+
+  Stage public doneStage;
+
+  SplitMerge private splitMerge;
+  SpaceToken private spaceToken;
+
+  address public subjectTokenOwner;
+  uint256 public subjectTokenId;
+  uint256[] public subjectContour;
+  uint256[] public clippingContour;
+
+  uint256[] public subjectContourOutput;
+  uint256[][] public resultContours;
+
+  constructor(address _spaceToken, address _subjectTokenOwner, uint256 _subjectTokenId, uint256[] _subjectContour, uint256[] _clippingContour) public {
+    splitMerge = SplitMerge(msg.sender);
+    spaceToken = SpaceToken(_spaceToken);
+
+    subjectTokenOwner = _subjectTokenOwner;
+    subjectTokenId = _subjectTokenId;
+    subjectContour = _subjectContour;
+    clippingContour = _clippingContour;
+  }
+
+  function init() public {
```

Comment:
It’s better to name a function external if it’s not supposed to be called from another method of the contract. This makes code analysis way more easier.